### ISSUE TITLE
Panic with OffsetFetchRequest

### ISSUE THREAD

**Issue by keitwb** [twmb/franz-go#317]
I'm using franz-go in some unit tests to watch produced and committed offsets and I'm occasionally seeing the same panic in our CI environment in tests (it occurs maybe 10% of the time or so):

```
panic: interface conversion: kmsg.Request is *kmsg.OffsetFetchRequest, not *kgo.pinReq [recovered]
	panic: interface conversion: kmsg.Request is *kmsg.OffsetFetchRequest, not *kgo.pinReq
goroutine 34 [running]:
testing.tRunner.func1.2({0x13619e0, 0xc0000fe480})
	/usr/local/go/src/testing/testing.go:1396 +0x372
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1399 +0x5f0
panic({0x13619e0, 0xc0000fe480})
	/usr/local/go/src/runtime/panic.go:890 +0x262
github.com/twmb/franz-go/pkg/kgo.(*Client).handleShardedReq.func2({0x0, {0x1666848, 0xc0000a0540}, {0x0, 0x0}})
	/go/pkg/mod/github.com/twmb/franz-go@v1.11.2/pkg/kgo/client.go:1804 +0xd05
github.com/twmb/franz-go/pkg/kgo.(*Client).handleShardedReq(0xc0005d2000, {0x1660918?, 0xc000036380}, {0x1666848?, 0xc0000a0540})
	/go/pkg/mod/github.com/twmb/franz-go@v1.11.2/pkg/kgo/client.go:1859 +0xed1
github.com/twmb/franz-go/pkg/kgo.(*Client).shardedRequest(0xc0005d2000, {0x1660950, 0xc00013a008}, {0x1666848, 0xc0000a0540})
	/go/pkg/mod/github.com/twmb/franz-go@v1.11.2/pkg/kgo/client.go:994 +0x811
github.com/twmb/franz-go/pkg/kgo.(*Client).Request(0xc0000add90?, {0x1660950, 0xc00013a008}, {0x1666848, 0xc0000a0540})
	/go/pkg/mod/github.com/twmb/franz-go@v1.11.2/pkg/kgo/client.go:778 +0x71
github.com/twmb/franz-go/pkg/kmsg.(*OffsetFetchRequest).RequestWith(...)
	/go/pkg/mod/github.com/twmb/franz-go/pkg/kmsg@v1.2.0/generated.go:10894
```

The code that runs the `RequestWith` call looks like this:

```
var resp *kmsg.OffsetFetchResponse
for {
  req := kmsg.NewOffsetFetchRequest()
  req.Group = c.ConsumerGroup
  c.mu.Lock()
  for tp := range c.offsets {
    req.Topics = append(req.Topics, kmsg.OffsetFetchRequestTopic{
      Topic:      tp.Topic,
      Partitions: []int32{tp.Partition},
    })
  }
  c.mu.Unlock()

  var err error
  resp, err = req.RequestWith(context.Background(), client) // <--- THIS LINE LEADS TO THE PANIC
  var ke *kerr.Error
  if err != nil && errors.As(err, &ke) {
    if ke.Code == 15 {
      time.Sleep(100 * time.Millisecond)
      continue
    }
  }
  require.Nil(tb, err)
  break
}
```

I'm on the latest version (1.11.2) on Go 1.19.  I'm being careful to not reuse the request object between tries.  Is this a bug or just user error?

Something that might be significant is that the unit tests use a single-broker kafka deployment.

**Comment by twmb** [comment]
This is a bug -- I'll patch asap. I remember testing exactly that scenario though so I'm not sure where I undid my original fix ... I'll also see how easy this is to wire into an integration test to avoid a repeat in the future.

**Comment by twmb** [comment]
I've recreated this, it can occur when fetching offsets for a group that doesn't exist, I see the fix.

**Comment by keitwb** [comment]
Ok thanks a ton.  What you described matches exactly the circumstances that can occur in the test environment.

**Comment by twmb** [comment]
Should be fixed in v1.11.3
